card_game.py

```diff
@@ -0,0 +1,214 @@
+        print("Your hand:")
+        for i, card in enumerate(self.hand_card, 1):
+            print(f"{i}. {card.name} (AP: {card.ap}) - {card.info}")
+
+
+class Enemy(Character):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.ap = 0  # Enemies have 0 action points
+        self.max_hand = 1
+
+
+class Game:
+    def __init__(self):
+        self.player = None
+        self.enemy = None
+
+    def create_characters(self):
+        # Initialize player and enemy characters
+        # The line below is removed since we initialize player and enemy later in the main function
+        # self.player = Player("Player", 30, deck=[Card(...), ...])  # Add player's cards
+        # self.enemy = Enemy("Enemy", 20, deck=[Card(...), ...])  # Add enemy's cards
+        pass
+
+    def display_info(self):
+        print("\n" + "-" * 30)
+        print(
+            f"Player HP: {self.player.current_hp}. Defense: {self.player.defense_point}")
+        print(
+            f"Enemy HP: {self.enemy.current_hp}. Defense: {self.enemy.defense_point}")
+        print("Enemy Intent:", self.enemy.hand_card[0].info)
+
+    def check_game_status(self):
+        if self.player.current_hp <= 0:
+            print("Game Over. You lost.")
+            return False
+
+        if self.enemy.current_hp <= 0:
+            print("You WIN!!")
+            return False
+
+        return True
+
+    def player_turn(self):
+        # Move remaining cards in hand to used_card
+        self.player.used_card.extend(self.player.hand_card)
+        self.player.hand_card.clear()
+
+        self.player.draw_additional_cards(self.player.max_hand)
+        self.player.ap = 3
+
+        while True:
+            self.display_info()
+            self.player.display_card_list()
+            print(f"Action Points: {self.player.ap}")
+
+            user_input = input(
+                "Choose a card to play (by number) or type 'END' to end your turn: ")
+            if user_input.upper() == "END":
+                break
+            try:
+                card_number = int(user_input) - 1
+                if 0 <= card_number < len(self.player.hand_card):
+                    self.player.play_card(
+                        self.player.hand_card[card_number], self.enemy)
+                    if not self.check_game_status():
+                        break
+                else:
+                    print("Invalid card number.")
+            except ValueError:
+                print(
+                    "Invalid input. Please enter a number or type 'END' to end your turn.")
+
+    def enemy_turn(self):
+        # Play exsisting hand_card (assuming only 1 card for enemy)
+        self.enemy.hand_card[0].play(self.enemy, self.player)
+
+        # Clean up Enemy card. Draw cards for the next round.
+        self.enemy.used_card.extend(self.enemy.hand_card)
+        self.enemy.hand_card.clear()
+        self.enemy.draw_additional_cards(1)
```

Comment:
nit: This can be removed if we move line 174 as the first line inside the `while True` loop.

Reply:
Done, thanks! 👍